hudi-client/hudi-java-client/src/test/java/org/apache/hudi/execution/bulkinsert/TestJavaBulkInsertInternalPartitioner.java

```diff
@@ -63,9 +65,11 @@ public void testCustomColumnSortPartitioner(String sortColumnString) throws Exce
         getCustomColumnComparator(HoodieTestDataGenerator.AVRO_SCHEMA, sortColumns);
 
     List<HoodieRecord> records = generateTestRecordsForBulkInsert(1000);
+    HoodieWriteConfig cfg = HoodieWriteConfig.newBuilder().withPath("basePath").build();
+    cfg.setValue(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME, "partition_path");
     testBulkInsertInternalPartitioner(
-        new JavaCustomColumnsSortPartitioner(sortColumns, HoodieTestDataGenerator.AVRO_SCHEMA, false),
-        records, true, generatePartitionNumRecords(records), Option.of(columnComparator));
+        new JavaCustomColumnsSortPartitioner(sortColumns, HoodieTestDataGenerator.AVRO_SCHEMA, false, cfg),
+        records, false, generatePartitionNumRecords(records), Option.of(columnComparator));
```

Comment:
Why did we flip this one to false?

Reply:
We actually shouldn't be flipping this boolean b/c records are still sorted -- just in a different way which is actually defined by the comparator above

Reply:
Let's also add/modify a regression test adding a check verifying number of files created (w/o your change there was a possibility that multiple files could be created w/in a single partition)

Reply:
Where should this test be located? I'm unsure on what level of the stack the test should be at so to know If I should just make the test in spark-sql for example or something more specific would help a lot